docs/configuration/jobs.rst

```diff
@@ -173,6 +173,74 @@ Note that you will have to trigger replication manually using the ``zrepl signal
        type: manual
      ...
 
+.. _job-snapshotting-hooks:
+
+Pre- and post-snapshot Hooks
+----------------------------
+
+Jobs with `periodic snapshots <job-snapshotting-spec_>`_ can run an executable before and/or after taking the snapshot,
+configured in an optional ``hooks`` section of the ``snapshotting`` configuration:
+
+::
+
+
+    jobs:
+    - type: push
+      filesystems: {
+        "<": true,
+        "tmp": false
+      }
+      snapshotting:
+        type: periodic
+        prefix: zrepl_
+        interval: 10m
+        hooks:
+          pre: scripts/pre-snapshot.sh
+          post: /path/to/scripts/zrepl-notify.sh
+          timeout: 30s
+          keep_inconsistent: false
+      ...
+
+The path to the hook executables must either be absolute (e.g. ``/opt/bin/zrepl-notify.sh``) or is relative to the current configuration
+file directory (e.g. ``scripts/pre-snapshot.sh`` will be found in ``/etc/zrepl/scripts/pre-snapshot.sh``).
+No arguments may be specified; create a wrapper script if zrepl must call an executable that requires arguments.
+The configuration may specify a ``pre`` script only, a ``post`` script only, or both.
+
+The ``timeout`` parameter is optional and specifies a period after which zrepl will kill the hook process and report an error.
+The default is 30 seconds and may be specified in any units understood by `time.ParseDuration <https://golang.org/pkg/time/#ParseDuration>`_.
+
+The ``keep_inconsistent`` parameter specifies whether a snapshot will be taken after a failed pre-snapshot hook. The default
+is ``false``, meaning the snapshot will be skipped in the pre-snapshot hook process exits with an error or the timeout is exceeded.
+Setting ``keep_inconsistent: true`` will allow snapshots to be taken even if the pre-snapshot hook fails, but could result in
+inconsistent snapshots if, for example, the pre-snapshot script is asking a database to write its state to disk.
+
+Zrepl sets a number of environment variables for the hook processes:
+
+* ``ZREPL_HOOKTYPE``: either "pre" or "post"
+* ``ZREPL_PHASE``: "Snapshotting"
+* ``ZREPL_FS``: the ZFS filesystem name being snapshotted
+* ``ZREPL_SNAPNAME``: the zrepl-generated snapshot name (e.g. ``zrepl_20380119_031407_000``)
```

Comment:
we should have `ZREPL_HOOK_DRYRUN=false|true` in here. There's no dry-run option ATM, but there could be in the future, and this is the only chance to get such a flag in without causing breakage in the future.
=> Also add this to the example hook.